EcommerceStore.Infrastructure/Repositories/BrandRepository.cs

```diff
@@ -0,0 +1,60 @@
+using EcommerceStore.Domain.Entities;
+using EcommerceStore.Domain.Interfaces;
+using EcommerceStore.Infrastucture.Persistence;
+using Microsoft.EntityFrameworkCore;
+
+namespace EcommerceStore.Infrastucture.Repositories
+{
+    public class BrandRepository : IBrandRepository
+    {
+        private readonly EcommerceContext _context;
+
+        public BrandRepository(EcommerceContext context)
+        {
+            _context = context;
+        }
+
+        public async Task CreateAsync(Brand brand)
+        {
+            await _context.Brands.AddAsync(brand);
+        }
+
+        public async Task<IEnumerable<Brand>> GetAllAsync()
+        {
+            var brands = await _context.Brands
+                .Include(b => b.Products)
+                .ToListAsync();
+
+            return brands;
+        }
+
+        public async Task<Brand> GetByIdAsync(int brandId)
```

Comment:
in case of get methods it is ok, when repo is responsible for mapping the result to view model(your previous implementation) and service is just calling repo method

Reply:
Alright. But the question is how am I going to return BrandViewModel, in this case, from BrandRepository, if the Domain layer doesn't know anything about BrandViewModel or BrandInputModel, because these models are located at the Application layer. So I can't define contract for these models. Should I move view/input models up to Domain layer?

Reply:
Let's keep your current implementation